OpenRA.Mods.Common/Effects/RallyPointIndicator.cs

```diff
@@ -97,7 +97,7 @@ IEnumerable<IRenderable> IEffectAboveShroud.RenderAboveShroud(WorldRenderer wr)
 
 		IEnumerable<IRenderable> RenderInner(WorldRenderer wr)
 		{
-			var palette = wr.Palette(rp.PaletteName);
+			var palette = circles != null && flag != null ? wr.Palette(rp.PaletteName) : null;
```

Comment:
Instead of defaulting an unused variable to an unused value, how about we change the code below the target line to something like:
```csharp
if (circles != null || flag != null)
{
	var palette = wr.Palette(rp.PaletteName);
	if (circles != null)
		// and so on...
```
?